eCommerce platforms/Adobe Magento/src/code/local/Microsoft/Dfp/Model/Observer.php

```diff
@@ -0,0 +1,251 @@
+				$this->dfp->log('************** Invoking UpdateAccount Address Change API - END ************* ');
+			}
+		} catch (exception $e) {
+			Mage::logException($e);
+		}
+	}
+
+	public function salesEventOrderAfterSave(Varien_Event_Observer $observer)
+	{
+		try {
+			$this->UpdateStatusToDFP($observer);
+		} catch (exception $e) {
+			Mage::logException($e);
+		}
+	}
+
+	private function UpdateStatusToDFP($observer)
+	{
+		$this->dfp->log("  -----------  UpdateStatusToDFP Observer Event START --------------  ");
+		$order = $observer->getEvent()->getOrder();
+		if ($order) {
+			if (!empty($order->getData()['status'])) {
+				$status = $order->getData()['status'];
+				if (!empty($status)) {
+					$this->dfp->log(" UpdateStatusToDFP Observer Event Status : " . $status);
+					switch (strtolower($status)) {
+						case 'order_approved':
+						case 'canceled':
+							$this->dfp->log("  -----------   Invoking PurchaseStatus API - START --------------  ");
+							$this->invokePurchaseStatusDFP($order);
+							$this->dfp->log("  -----------   Invoking PurchaseStatus API - END --------------  ");
+							break;
+						case 'refunded':
+							$this->dfp->log("  -----------   Invoking Refund API - START --------------  ");
+							$this->invokeRefundDFP($order);
+							$this->dfp->log("  -----------   Invoking Refund API - END --------------  ");
+							break;
+						case 'chargeback':
+							$this->dfp->log("  -----------   Invoking Chargeback API - START --------------  ");
+							$this->invokeChargebackDFP($order);
+							$this->dfp->log("  -----------   Invoking Chargeback API - END --------------  ");
+							break;
+					}
+				}
+			}
+		}
+
+		$this->dfp->log("  -----------  UpdateStatusToDFP Observer Event END --------------  ");
+	}
+
+	private function invokeRefundDFP($refund)
+	{
+		$payload = $this->generateRefundPayload($refund);
+		return $this->dfp->sendRefund($payload);
+	}
+
+	private function invokeChargebackDFP($order)
+	{
+		$payload = $this->generateChargebackPayload($order);
+		return $this->dfp->sendChargeback($payload);
+	}
+
+	private function invokePurchaseStatusDFP($order)
+	{
+		$payload = $this->generatePurchaseStatusPayload($order);
+		return $this->dfp->sendPurchaseStatus($payload);
+	}
+
+	private function invokeUpdateAccountDFP($customer)
+	{
+		$payload = $this->generateUpdateAccountPayload($customer);
+		return $this->dfp->sendUpdateAccount($payload);
+	}
+
+	private function generatePurchaseStatusPayload($order)
+	{
+		$_metadata = array(
+			"trackingId"		=> $this->dfp->GUID(),
+			"merchantTimeStamp"	=> date('c')
+		);
+
+		$status = $order->getData()['status'];
+		if ($status == 'order_approved') {
+			$status = 'Approved';
+		} else {
+			$status = 'Canceled';
+		}
+
+		$purchaseStatusPayload = array(
+			'purchaseId'	=> $order->getData()['increment_id'],
+			'statusType'	=> $status,
+			'statusDate'	=> date('c'),
+			'reason'	    => '',
+			'_metadata'		=> $_metadata
+
```

Comment:
nit: lines 135 and 136 should be swapped :)

Reply:
or at least the empty line removed to match other methods

Reply:
Thanks!